### DISCUSSION TITLE
How can I set response headers?

### DISCUSSION THREAD

**Discussion by rimu** [plangrid/flask-rebar#236] [Q&A] [answered]
I've got a medium sized API all up and running, but one of the frontend devs is claiming the browser is caching the response from one of the endpoints. So I'd like to set the Cache-Control header.

Flask-rebar abstracts away the process of creating a response so I see no way to set the headers in it. I see [flask-rebar has a response function](https://flask-rebar.readthedocs.io/en/latest/api_reference.html#helpers) but attempts to return response(data) from my endpoint function have been unsuccessful.

Any tips?

**Comment by barakalon** [comment, marked as answer]
Ah, this is indeed lacking in the documentation.

You can return a tuple of `(response, status, headers)` or `(response, headers)` from a view function, similar to Flask view functions: https://flask.palletsprojects.com/en/2.0.x/quickstart/#about-responses

Here’s the source code, if you want to see exactly how the response is interpreted: https://github.com/plangrid/flask-rebar/blob/master/flask_rebar/rebar.py#L78

Hope that helps.

**Reply by rimu** [reply]
Ah, so something like

```
headers = {'Cache-Control': 'no-store, max-age=0'}
return (final_result, 200, headers)
```

Thanks!

**Comment by RookieRick** [comment]
Thanks for the quick field of that question Barak!  I'll add a ticket to shore up the documentation (and, with any luck, I'm actually going to have some time to work on Rebar next week in the slack tide between monster projects and Hackweek 🤣 )
